modules/apps/frontend-js/frontend-js-item-selector-web/src/main/resources/META-INF/resources/item_selector/ItemSelectorModal.tsx

```diff
@@ -102,28 +108,11 @@ function ItemSelectorModal<T extends Record<string, any>>({
 			<ClayModal.Body className="p-0">
 				<FrontendDataSet
 					{...fdsProps}
-					onSelect={({
-						selectedItems: newSelectedItems,
-					}: {
-						selectedItems: T[];
-					}) => {
-						if (
-							fdsProps.selectionType === 'single' &&
-							newSelectedItems.length > 1
-						) {
-							setSelectedItems(newSelectedItems.slice(0, 1));
-						}
-						else {
-							setSelectedItems(newSelectedItems);
-						}
-					}}
-					selectedItems={selectedItems.map((item) =>
-						getObjectValueFromPath({
-							object: item,
-							path: locator.value,
-						})
-					)}
-					selectedItemsKey={locator.id}
+					onSelectedItemsChange={setSelectedItems}
+					selectedItems={selectedItems}
+					selectedItemsKey={locator.value}
```

Comment:
By moving the calculation of selected items value to FDS, we do need to pass `locator.value` to it, but then it makes `locator.id` useless for here.

As @ethib137 mentioned, we want to keep a similar API between `ItemSelector` and `ItemSelectorModal`, but `ItemSelector` does uses `locator.id`. That makes me think, what is the use case we have there that makes it necessary, but we don't have here?

If it is really not necessary, maybe we should consider removing `id`, or `value`, from `locator`.

LMK what you think, @ethib137 and @bryceosterhaus.